packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -500,7 +500,8 @@ class ReactShallowRenderer {
       element.type,
     );
     invariant(
-      isForwardRef(element) || typeof element.type === 'function',
+      isForwardRef(element) ||
+        (typeof element.type === 'function' || isMemo(element.type)),
```

Comment:
any reason not to use isValidElementType from react-is here, and asserting the type isn’t string, rather than repeating some of the logic?

Reply:
Probably because `isValidElementType` was introduced after this version of the shallow renderer. That's a good point though, because looking at `isValidElementType` catches more uncommon cases that React technically supports and that fail with the shallow renderer.

For example, [you can do](https://codesandbox.io/s/vqz0ryn7xy):

```js
const MemoFragment = React.memo(React.Fragment)
```

But rendering that with the shallow render will trigger this invariant. The same goes for other elements that have their own type like `React.Suspense`, `React.ConcurrentMode`, `React.StrictMode`, etc.

After this is approved I'll submit a follow up that switches to `isValidElementType` and adds test for all those cases.

Reply:
Though it seems unlikely that anyone would be shallow rendering `React.memo(React.Fragment)` 😅